backend/app/crud/crud_role.py

```diff
@@ -37,25 +37,20 @@ async def get_by_name(self, db, name: str) -> Role | None:
 
     async def create(self, db, obj_in: CreateRole) -> NoReturn:
         new_role = self.model(**obj_in.dict(exclude={'menus'}))
-        menu_list = []
-        for menu_id in obj_in.menus:
-            menu_list.append(await db.get(Menu, menu_id))
-        new_role.menus.append(*menu_list)
+        menus = await db.execute(select(Menu).where(Menu.id.in_(obj_in.menus)))
+        new_role.menus = menus.scalars().all()
```

Comment:
`append()` seems to be the recommended method in the official documentation

https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#combining-association-object-with-many-to-many-access-patterns

Reply:
Yes, I know, but there will be exceptions when using traversal remove and append, and it needs to be solved.

Here, one "in" query can be used to achieve the effect, the test is normal, and the intermediate table will be updated and deleted accordingly.

Reply:
Exceptions don't happen because logical judgments are made in the business code?

https://github.com/fastapi-practices/fastapi_best_architecture/blob/01fb3919349f5d84b44ee47435877e7d27be9ca8/backend/app/services/role_service.py#L32-L35

If it works, it certainly looks cleaner this way.